### DISCUSSION TITLE
How can I disable a user in supabase auth database?

### DISCUSSION THREAD

**Discussion by ADSNB** [supabase/supabase#26771] [Questions] [answered]
I`m using supabase auth for my app`s authentication.

I need to disable a user for an unlimited time and this user should not be able to login in the app anymore. How can I do that with supabase auth?

I don`t wanna delete a user since I have table records attached to the supabase users auth table.

**Comment by GaryAustin1** [comment]
The only built in option is set the ban_duration with a very long time.

I prefer the approach of all my data tables' foreign keys pointing to a profile table that is created when a user is created.  There is no fk to auth.users from the profile table.   Then you have an auth.users delete trigger clean out the profile row and mark the user as guest or some such.

**Reply by GaryAustin1** [reply]
I forgot about the ability to soft delete...
https://supabase.com/docs/reference/javascript/auth-admin-deleteuser
There is a soft delete option.

**Reply by ADSNB** [reply]
> I forgot about the ability to soft delete... https://supabase.com/docs/reference/javascript/auth-admin-deleteuser There is a soft delete option.

I tried the soft delete option. Unfortunately when we use this option supabase automatically delete user displayname, email and phone number.

I guess I should stick to the ban duration that you mentioned for now with a huge value in place like thousands of years, hehehe.

It`s a shame we don`t have this functionality yet in supabase auth because it`s a very common feature nowadays though.

**Reply by GaryAustin1** [reply, marked as answer]
Seems like ban is what you want.  Soft delete just means the user id stays so existing records don't have to be deleted if linked to auth.users.  It does not mean the user gets to reclaim them again

**Reply by ADSNB** [reply]
> Seems like ban is what you want. Soft delete just means the user id stays so existing records don't have to be deleted if linked to auth.users. It does not mean the user gets to reclaim them again

Got it. 

Yeap, I will use ban duration for now. Still it will be a good feature we have the option to have a boolean attribute like IsActive in user auth table, so we can change the flag and doing so user will be allowed to login or not in the app. Because ban duration still is a temporary ban, sometimes the manager wants to disable a user but they don`t know when the user will be activated again. 

I will probably request a new feature for this.

Anyway, thank you for your help! Much appreciated!
